src-java/kilda-pce/src/main/java/org/openkilda/pce/model/PathWeight.java

```diff
@@ -15,78 +15,108 @@
 
 package org.openkilda.pce.model;
 
-import static com.google.common.primitives.Longs.asList;
+import lombok.ToString;
 
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.Map;
 
 /**
  * Vector path weight representation. Each value in vector corresponds to some path param defined
  * by {@link WeightFunction}. Only PathWeights created by one WeightFunction should be added and compared.
  */
+@ToString
 public class PathWeight implements Comparable<PathWeight> {
 
-    private List<Long> params;
+    private final long baseWeight;
+    private final long alternativeBaseWeight;
```

Comment:
I don't know a better name for it ;)
In case of "weightByCostAndAvailableBandwidth" we have to hold alternative parameter for correct comparation.

Please see the compareTo-method:
First of all, it checks `getTotalWeight` (baseWeight + all penalties, but without alternativeBaseWeight)
and if both weights are the same, then it checks `alternativeBaseWeight` (without baseWeight and penalties).

This behaviour checked by tests and I just reproduce this logic in term of new words.